### PR TITLE
refactor(uuid): specify the stream node uuid when using grpc connection

### PR DESCRIPTION
**Pull request by rgallor**
Use EVN ASTARTE_MSGHUB_NODE_ID or the toml file to specify the node id of the stream application

### CODE REVIEW

**Review by harlem88 — APPROVED** [review]

File: src/astarte.rs [hunk 1]
```diff
@@ -103,9 +101,10 @@ impl ConnectionConfigBuilder {
                 self.astarte_connection = Some(con);
 
                 let endpoint = env::var("ASTARTE_MSGHUB_ENDPOINT")?;
+                let node_id = env::var("ASTARTE_MSGHUB_NODE_ID")?.parse::<Uuid>()?;
```
[the remark this answers is not in the document]
Comment (rgallor) [reply src/astarte.rs:104]: Yes it could be a good idea

**Review by lucaato — APPROVED** [review]

File: src/astarte.rs [hunk 2]
```diff
@@ -103,9 +103,12 @@ impl ConnectionConfigBuilder {
                 self.astarte_connection = Some(con);
 
                 let endpoint = env::var("ASTARTE_MSGHUB_ENDPOINT")?;
+                let node_id = env::var("ASTARTE_MSGHUB_NODE_ID")
+                    .map(|s| s.parse::<Uuid>().unwrap_or(DEFAULT_STREAM_NODE_ID))
```
Comment (joshuachp) [inline src/astarte.rs:107]:
```suggestion
                let node_id = match env::var("ASTARTE_MSGHUB_NODE_ID") {
                    Ok(uuid) => Uuid::parse_str(uuid).wrap_err("invalid ASTARTE_MSGHUB_NODE_ID")?;
                    Err(VarError::NotPresent) => DEFAULT_STREAM_NODE_ID,
                    Err(VarError::NotUnicode) => bail!("non unicode ASTARTE_MSGHUB_NODE_ID"),
                };
```

**Review by joshuachp — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -62,6 +63,7 @@ astarte_ignore_ssl = false
 # gRPC connection to the Astarte Message Hub
 [astarte.grpc]
 endpoint = "http://[::1]:50051"
+node_id = "ASTARTE_MSGHUB_NODE_ID_HERE"
```
Comment (joshuachp) [inline README.md:66]:
```suggestion
#node_id = "ASTARTE_MSGHUB_NODE_ID_HERE"
```

**Review by joshuachp — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (rgallor) [reply README.md:66]: Do you think the stream node id should be optional also when retrieving the configuration from the config.toml?

Comment (joshuachp) [reply README.md:66]: O yes, I missed the Deserialize bound on the builder. I think an overridable serde(default = "...") would be more user-friendly

File: astarte-device-conf/config.toml [hunk 4]
```diff
@@ -6,17 +6,17 @@
...
+#[astarte]
+#connection = "mqtt"
+#store_directory = "/tmp/stream-rust-test/store/"
+#
+#[astarte.mqtt]
+#realm = "test"
+#device_id = "2TBn-jNESuuHamE2Zo1anA"
+#pairing_url = "http://api.astarte.localhost/pairing"
+#credentials_secret = "76IsmXPTDTOhw4H/cZW+lefFRMjIQxPyYOIhLSX3REU="
+##pairing_token = "PAIRING_TOKEN_HERE"
```
Comment (joshuachp) [inline astarte-device-conf/config.toml:18]:
```suggestion
#pairing_token = "PAIRING_TOKEN_HERE"
```

**Review by joshuachp — APPROVED** [review]

**Review by harlem88 — APPROVED** [review]

**Pull request merged**